plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/version/KubernetesVersionManagerImpl.java

```diff
@@ -0,0 +1,377 @@
+        response.setName(kubernetesSupportedVersion.getName());
+        response.setSemanticVersion(kubernetesSupportedVersion.getSemanticVersion());
+        DataCenterVO zone = dataCenterDao.findById(kubernetesSupportedVersion.getZoneId());
+        if (zone != null) {
+            response.setZoneId(zone.getUuid());
+            response.setZoneName(zone.getName());
+        }
+        if (compareSemanticVersions(kubernetesSupportedVersion.getSemanticVersion(),
+                KubernetesClusterService.MIN_KUBERNETES_VERSION_HA_SUPPORT)>=0) {
+            response.setSupportsHA(true);
+        } else {
+            response.setSupportsHA(false);
+        }
+        TemplateJoinVO template = templateJoinDao.findById(kubernetesSupportedVersion.getIsoId());
+        response.setIsoId(template.getUuid());
+        response.setIsoName(template.getName());
+        response.setIsoState(template.getState().toString());
+        return response;
+    }
+
+    private ListResponse<KubernetesSupportedVersionResponse> createKubernetesSupportedVersionListResponse(List<KubernetesSupportedVersionVO> versions) {
+        List<KubernetesSupportedVersionResponse> responseList = new ArrayList<>();
+        for (KubernetesSupportedVersionVO version : versions) {
+            responseList.add(createKubernetesSupportedVersionResponse(version));
+        }
+        ListResponse<KubernetesSupportedVersionResponse> response = new ListResponse<>();
+        response.setResponses(responseList);
+        return response;
+    }
+
+    private static boolean isSemanticVersion(final String version) {
+        if(!version.matches("[0-9]+(\\.[0-9]+)*")) {
+            return false;
+        }
+        String[] parts = version.split("\\.");
+        if (parts.length < 3) {
+            return false;
+        }
+        return true;
+    }
+
+    public static int compareSemanticVersions(String v1, String v2) throws IllegalArgumentException {
+        if (Strings.isNullOrEmpty(v1) || Strings.isNullOrEmpty(v2)) {
+            throw new IllegalArgumentException(String.format("Invalid version comparision with versions %s, %s", v1, v2));
+        }
+        if(!isSemanticVersion(v1)) {
+            throw new IllegalArgumentException(String.format("Invalid version format, %s", v1));
+        }
+        if(!isSemanticVersion(v2)) {
+            throw new IllegalArgumentException(String.format("Invalid version format, %s", v2));
+        }
+        String[] thisParts = v1.split("\\.");
+        String[] thatParts = v2.split("\\.");
+        int length = Math.max(thisParts.length, thatParts.length);
+        for(int i = 0; i < length; i++) {
+            int thisPart = i < thisParts.length ?
+                    Integer.parseInt(thisParts[i]) : 0;
+            int thatPart = i < thatParts.length ?
+                    Integer.parseInt(thatParts[i]) : 0;
+            if(thisPart < thatPart)
+                return -1;
+            if(thisPart > thatPart)
+                return 1;
```

Comment:
shall we return thisPart-thatPart, so the caller could know the difference if more than 1s?

Reply:
Still, the caller won't be able to know if the difference is for major version or minor version or for the patch so won't be much useful

Reply:
you are right, it would however reduce the code ;)